### ISSUE TITLE
dev version build / install issues

### ISSUE THREAD

**Issue by stas00** [jupyter/nbdime#391]
Trying to use your unreleased version, I tried to build a dev env as explained here using python 3.6.5:
https://nbdime.readthedocs.io/en/latest/installing.html#installing-latest-development-version

I think I figured out all the issues, and posting the below for the benefit of other users who might have similar issues, and perhaps some of these could be added to the docs if you feel it fits.

1) I didn't have venv and had to install it (it won't install it via pip):

`apt-get install python3-venv`

2) Initially, I couldn't get past:

```
python3.6 -m venv myenv
Error: Command '['/home/stas/python-dev/myenv/bin/python3.6', '-Im', 'ensurepip', '--upgrade', '--default-pip']' returned non-zero exit status 1.
```

Googling I found that ubuntu's python package doesn't include `ensurepip`, so I had to manually add it:

```
wget https://www.python.org/ftp/python/3.6.5/Python-3.6.5.tgz
tar -xvzf Python-3.6.5.tgz
cp -r ./Python-3.6.5/Lib/ensurepip /usr/lib/python3.6/
```

Then I was able to complete the build. 

3) Another note: The build won't work as root, only as user.

4) Finally, how do I now invoke jupyter notebook to use this custom environment? I'm new to virtualenv - reading the docs on it, it seems that the deployment should be:

```
cd ~/python-dev
python3.6 -m venv myenv
source myenv/bin/activate
jupyter notebook
```

and perhaps a link to: https://docs.python.org/3/library/venv.html

p.s. perhaps the docs can be modified to include the creation of a directory where the new environment is built, so that it can be shown in the deployment steps as I have shown above. e.g.:

```
mkdir ~/python-dev
cd ~/python-dev
python3 -m venv myenv          # For Python 2: python2 -m virtualenv myenv
source myenv/bin/activate
pip install nodeenv
nodeenv -p
```

Thank you.

**Comment by vidartf** [comment]
Those instructions related to venv/node are mainly for putting the typescript development in an environment. If all you want to do is install master, use:

```
pip install git+https://github.com/jupyter/nbdime#egg=nbdime
```

You will need to have node installed though (`conda install nodejs` should be enough I think).

If you want to tweak the python code, try this:

```
git clone https://github.com/jupyter/nbdime
cd nbdime
pip install -e .
```

I'll try to work a little on the dev install instructions to make them a little less opinionated (venv is just one of many ways to solve the problem of node).

**Comment by stas00** [comment]
Could you please add to the top-level doc to do:

`pip install nodejs `

before:

`pip install git+https://github.com/jupyter/nbdime#egg=nbdime`

Because currently it says:

`To install a development version of nbdime, you will need npm installed and available on your PATH while installing.`

and when I followed the instruction to install npm, it won't build from git, with a bunch of errors from learna about webkit and stuff.

Thank you!

**Comment by stas00** [comment]
But no, it won't build it

```
pip3 uninstall nbdime
pip3 install git+https://github.com/jupyter/nbdime#egg=nbdime
[...]
Building wheels for collected packages: nbdime
  Running setup.py bdist_wheel for nbdime ... error
  Complete output from command /usr/bin/python3.6 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-cg86t734/nbdime/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/pip-wheel-z2c79wjq --python-tag cp36:
  /usr/local/lib/python3.6/dist-packages/setuptools/dist.py:398: UserWarning: Normalizing '1.0.2.dev' to '1.0.2.dev0'
    normalized_version,
  running bdist_wheel
  running js
  Installing build dependencies with npm.  This may take a while...
  > npm install
  npm WARN lifecycle undefined~install: cannot run in wd %s %s (wd=%s) undefined lerna bootstrap --hoist /tmp/pip-install-cg86t734/nbdime
  /tmp/pip-install-cg86t734/nbdime
  ├─┬ lerna@2.11.0
  │ ├── async@1.5.2
  │ ├─┬ chalk@2.4.1
  │ │ ├─┬ ansi-styles@3.2.1
  │ │ │ └─┬ color-convert@1.9.2
  │ │ │   └── color-name@1.1.1
  │ │ ├── escape-string-regexp@1.0.5
  │ │ └─┬ supports-color@5.4.0
  │ │   └── has-flag@3.0.0
  │ ├─┬ cmd-shim@2.0.2
  │ │ └─┬ mkdirp@0.5.1
  │ │   └── minimist@0.0.8
  │ ├─┬ columnify@1.5.4
  │ │ ├─┬ strip-ansi@3.0.1
  │ │ │ └── ansi-regex@2.1.1
  │ │ └─┬ wcwidth@1.0.1
  │ │   └─┬ defaults@1.0.3
  │ │     └── clone@1.0.4
  │ ├── command-join@2.0.0
  │ ├─┬ conventional-changelog-cli@1.3.22
  │ │ ├── add-stream@1.0.0
  │ │ ├─┬ conventional-changelog@1.1.24
  │ │ │ ├─┬ conventional-changelog-angular@1.6.6
  │ │ │ │ ├─┬ compare-func@1.3.2
  │ │ │ │ │ ├── array-ify@1.0.0
  │ │ │ │ │ └─┬ dot-prop@3.0.0
  │ │ │ │ │   └── is-obj@1.0.1
  │ │ │ │ └── q@1.5.1
  │ │ │ ├── conventional-changelog-atom@0.2.8
  │ │ │ ├── conventional-changelog-codemirror@0.3.8
  │ │ │ ├─┬ conventional-changelog-core@2.0.11
  │ │ │ │ ├─┬ conventional-changelog-writer@3.0.9
  │ │ │ │ │ ├─┬ handlebars@4.0.11
  │ │ │ │ │ │ ├─┬ optimist@0.6.1
  │ │ │ │ │ │ │ └── wordwrap@0.0.3
  │ │ │ │ │ │ ├─┬ source-map@0.4.4
  │ │ │ │ │ │ │ └── amdefine@1.0.1
  │ │ │ │ │ │ └─┬ uglify-js@2.8.29
  │ │ │ │ │ │   ├── source-map@0.5.7
  │ │ │ │ │ │   ├── uglify-to-browserify@1.0.2
  │ │ │ │ │ │   └─┬ yargs@3.10.0
  │ │ │ │ │ │     ├── camelcase@1.2.1
  │ │ │ │ │ │     ├─┬ cliui@2.1.0
  │ │ │ │ │ │     │ ├─┬ center-align@0.1.3
  │ │ │ │ │ │     │ │ ├─┬ align-text@0.1.4
  │ │ │ │ │ │     │ │ │ ├─┬ kind-of@3.2.2
  │ │ │ │ │ │     │ │ │ │ └── is-buffer@1.1.6
  │ │ │ │ │ │     │ │ │ ├── longest@1.0.1
  │ │ │ │ │ │     │ │ │ └── repeat-string@1.6.1
  │ │ │ │ │ │     │ │ └── lazy-cache@1.0.4
  │ │ │ │ │ │     │ ├── right-align@0.1.3
  │ │ │ │ │ │     │ └── wordwrap@0.0.2
  │ │ │ │ │ │     └── window-size@0.1.0
  │ │ │ │ │ ├── json-stringify-safe@5.0.1
  │ │ │ │ │ └── split@1.0.1
  │ │ │ │ ├── dateformat@3.0.3
  │ │ │ │ ├─┬ get-pkg-repo@1.4.0
  │ │ │ │ │ ├─┬ meow@3.7.0
  │ │ │ │ │ │ ├─┬ camelcase-keys@2.1.0
  │ │ │ │ │ │ │ └── camelcase@2.1.1
  │ │ │ │ │ │ ├── map-obj@1.0.1
  │ │ │ │ │ │ ├── minimist@1.2.0
  │ │ │ │ │ │ ├─┬ redent@1.0.0
  │ │ │ │ │ │ │ ├── indent-string@2.1.0
  │ │ │ │ │ │ │ └── strip-indent@1.0.1
  │ │ │ │ │ │ └── trim-newlines@1.0.0
  │ │ │ │ │ └── parse-github-repo-url@1.4.1
  │ │ │ │ ├─┬ git-remote-origin-url@2.0.0
  │ │ │ │ │ ├── gitconfiglocal@1.0.0
  │ │ │ │ │ └── pify@2.3.0
  │ │ │ │ └─┬ read-pkg@1.1.0
  │ │ │ │   ├─┬ load-json-file@1.1.0
  │ │ │ │   │ ├── parse-json@2.2.0
  │ │ │ │   │ ├── pify@2.3.0
  │ │ │ │   │ └─┬ strip-bom@2.0.0
  │ │ │ │   │   └── is-utf8@0.2.1
  │ │ │ │   └── path-type@1.1.0
  │ │ │ ├── conventional-changelog-ember@0.3.12
  │ │ │ ├── conventional-changelog-eslint@1.0.9
  │ │ │ ├── conventional-changelog-express@0.3.6
  │ │ │ ├── conventional-changelog-jquery@0.1.0
  │ │ │ ├── conventional-changelog-jscs@0.1.0
  │ │ │ ├── conventional-changelog-jshint@0.3.8
  │ │ │ └── conventional-changelog-preset-loader@1.1.8
  │ │ ├─┬ meow@4.0.1
  │ │ │ ├─┬ camelcase-keys@4.2.0
  │ │ │ │ ├── camelcase@4.1.0
  │ │ │ │ ├── map-obj@2.0.0
  │ │ │ │ └── quick-lru@1.1.0
  │ │ │ ├─┬ decamelize-keys@1.1.0
  │ │ │ │ └── map-obj@1.0.1
  │ │ │ ├─┬ loud-rejection@1.6.0
  │ │ │ │ └─┬ currently-unhandled@0.4.1
  │ │ │ │   └── array-find-index@1.0.2
  │ │ │ ├── minimist@1.2.0
  │ │ │ ├─┬ minimist-options@3.0.2
  │ │ │ │ └── arrify@1.0.1
  │ │ │ ├── read-pkg-up@3.0.0
  │ │ │ ├─┬ redent@2.0.0
  │ │ │ │ ├── indent-string@3.2.0
  │ │ │ │ └── strip-indent@2.0.0
  │ │ │ └── trim-newlines@2.0.0
  │ │ └─┬ tempfile@1.1.1
  │ │   ├── os-tmpdir@1.0.2
  │ │   └── uuid@2.0.3
  │ ├─┬ conventional-recommended-bump@1.2.1
  │ │ ├─┬ concat-stream@1.6.2
  │ │ │ ├── buffer-from@1.1.0
  │ │ │ ├─┬ readable-stream@2.3.6
  │ │ │ │ ├── core-util-is@1.0.2
  │ │ │ │ ├── isarray@1.0.0
  │ │ │ │ ├── process-nextick-args@2.0.0
  │ │ │ │ ├── string_decoder@1.1.1
  │ │ │ │ └── util-deprecate@1.0.2
  │ │ │ └── typedarray@0.0.6
  │ │ ├─┬ conventional-commits-filter@1.1.6
  │ │ │ ├── is-subset@0.1.1
  │ │ │ └── modify-values@1.0.1
  │ │ ├─┬ conventional-commits-parser@2.1.7
  │ │ │ ├─┬ is-text-path@1.0.1
  │ │ │ │ └── text-extensions@1.7.0
  │ │ │ ├─┬ JSONStream@1.3.3
  │ │ │ │ └── jsonparse@1.3.1
  │ │ │ ├── split2@2.2.0
  │ │ │ ├─┬ through2@2.0.3
  │ │ │ │ └── xtend@4.0.1
  │ │ │ └── trim-off-newlines@1.0.1
  │ │ ├─┬ git-raw-commits@1.3.6
  │ │ │ ├─┬ dargs@4.1.0
  │ │ │ │ └── number-is-nan@1.0.1
  │ │ │ └─┬ lodash.template@4.4.0
  │ │ │   ├── lodash._reinterpolate@3.0.0
  │ │ │   └── lodash.templatesettings@4.1.0
  │ │ ├── git-semver-tags@1.3.6
  │ │ ├─┬ meow@3.7.0
  │ │ │ ├─┬ camelcase-keys@2.1.0
  │ │ │ │ └── camelcase@2.1.1
  │ │ │ ├── map-obj@1.0.1
  │ │ │ ├── minimist@1.2.0
  │ │ │ ├─┬ read-pkg-up@1.0.1
  │ │ │ │ ├─┬ find-up@1.1.2
  │ │ │ │ │ └── path-exists@2.1.0
  │ │ │ │ └─┬ read-pkg@1.1.0
  │ │ │ │   ├─┬ load-json-file@1.1.0
  │ │ │ │   │ ├── parse-json@2.2.0
  │ │ │ │   │ ├── pify@2.3.0
  │ │ │ │   │ └── strip-bom@2.0.0
  │ │ │ │   └── path-type@1.1.0
  │ │ │ ├─┬ redent@1.0.0
  │ │ │ │ ├─┬ indent-string@2.1.0
  │ │ │ │ │ └─┬ repeating@2.0.1
  │ │ │ │ │   └── is-finite@1.0.2
  │ │ │ │ └─┬ strip-indent@1.0.1
  │ │ │ │   └── get-stdin@4.0.1
  │ │ │ └── trim-newlines@1.0.0
  │ │ └── object-assign@4.1.1
  │ ├── dedent@0.7.0
  │ ├─┬ execa@0.8.0
  │ │ ├─┬ cross-spawn@5.1.0
  │ │ │ ├─┬ lru-cache@4.1.3
  │ │ │ │ ├── pseudomap@1.0.2
  │ │ │ │ └── yallist@2.1.2
  │ │ │ ├─┬ shebang-command@1.2.0
  │ │ │ │ └── shebang-regex@1.0.0
  │ │ │ └─┬ which@1.3.1
  │ │ │   └── isexe@2.0.0
  │ │ ├── get-stream@3.0.0
  │ │ ├── is-stream@1.1.0
  │ │ ├─┬ npm-run-path@2.0.2
  │ │ │ └── path-key@2.0.1
  │ │ └── strip-eof@1.0.0
  │ ├─┬ find-up@2.1.0
  │ │ └─┬ locate-path@2.0.0
  │ │   └─┬ p-locate@2.0.0
  │ │     └─┬ p-limit@1.3.0
  │ │       └── p-try@1.0.0
  │ ├─┬ fs-extra@4.0.3
  │ │ ├── jsonfile@4.0.0
  │ │ └── universalify@0.1.1
  │ ├── get-port@3.2.0
  │ ├─┬ glob@7.1.2
  │ │ ├── fs.realpath@1.0.0
  │ │ ├─┬ inflight@1.0.6
  │ │ │ └── wrappy@1.0.2
  │ │ ├── inherits@2.0.3
  │ │ ├── once@1.4.0
  │ │ └── path-is-absolute@1.0.1
  │ ├─┬ glob-parent@3.1.0
  │ │ ├─┬ is-glob@3.1.0
  │ │ │ └── is-extglob@2.1.1
  │ │ └── path-dirname@1.0.2
  │ ├─┬ globby@6.1.0
  │ │ ├─┬ array-union@1.0.2
  │ │ │ └── array-uniq@1.0.3
  │ │ ├── pify@2.3.0
  │ │ └─┬ pinkie-promise@2.0.1
  │ │   └── pinkie@2.0.4
  │ ├── graceful-fs@4.1.11
  │ ├── hosted-git-info@2.6.0
  │ ├─┬ inquirer@3.3.0
  │ │ ├── ansi-escapes@3.1.0
  │ │ ├─┬ cli-cursor@2.1.0
  │ │ │ └─┬ restore-cursor@2.0.0
  │ │ │   └── onetime@2.0.1
  │ │ ├── cli-width@2.2.0
  │ │ ├─┬ external-editor@2.2.0
  │ │ │ ├── chardet@0.4.2
  │ │ │ ├─┬ iconv-lite@0.4.23
  │ │ │ │ └── safer-buffer@2.1.2
  │ │ │ └── tmp@0.0.33
  │ │ ├── figures@2.0.0
  │ │ ├── mute-stream@0.0.7
  │ │ ├─┬ run-async@2.3.0
  │ │ │ └── is-promise@2.1.0
  │ │ ├── rx-lite@4.0.8
  │ │ ├── rx-lite-aggregates@4.0.8
  │ │ ├─┬ string-width@2.1.1
  │ │ │ ├── is-fullwidth-code-point@2.0.0
  │ │ │ └─┬ strip-ansi@4.0.0
  │ │ │   └── ansi-regex@3.0.0
  │ │ ├─┬ strip-ansi@4.0.0
  │ │ │ └── ansi-regex@3.0.0
  │ │ └── through@2.3.8
  │ ├─┬ is-ci@1.1.0
  │ │ └── ci-info@1.1.3
  │ ├─┬ load-json-file@4.0.0
  │ │ ├─┬ parse-json@4.0.0
  │ │ │ ├─┬ error-ex@1.3.1
  │ │ │ │ └── is-arrayish@0.2.1
  │ │ │ └── json-parse-better-errors@1.0.2
  │ │ ├── pify@3.0.0
  │ │ └── strip-bom@3.0.0
  │ ├── lodash@4.17.10
  │ ├─┬ minimatch@3.0.4
  │ │ └─┬ brace-expansion@1.1.11
  │ │   ├── balanced-match@1.0.0
  │ │   └── concat-map@0.0.1
  │ ├─┬ npmlog@4.1.2
  │ │ ├─┬ are-we-there-yet@1.1.5
  │ │ │ └── delegates@1.0.0
  │ │ ├── console-control-strings@1.1.0
  │ │ ├─┬ gauge@2.7.4
  │ │ │ ├── aproba@1.2.0
  │ │ │ ├── has-unicode@2.0.1
  │ │ │ ├─┬ string-width@1.0.2
  │ │ │ │ ├── code-point-at@1.1.0
  │ │ │ │ └── is-fullwidth-code-point@1.0.0
  │ │ │ └── wide-align@1.1.3
  │ │ └── set-blocking@2.0.0
  │ ├── p-finally@1.0.0
  │ ├─┬ package-json@4.0.1
  │ │ ├─┬ got@6.7.1
  │ │ │ ├─┬ create-error-class@3.0.2
  │ │ │ │ └── capture-stack-trace@1.0.0
  │ │ │ ├── duplexer3@0.1.4
  │ │ │ ├── is-redirect@1.0.0
  │ │ │ ├── is-retry-allowed@1.1.0
  │ │ │ ├── lowercase-keys@1.0.1
  │ │ │ ├── timed-out@4.0.1
  │ │ │ ├── unzip-response@2.0.1
  │ │ │ └─┬ url-parse-lax@1.0.0
  │ │ │   └── prepend-http@1.0.4
  │ │ ├─┬ registry-auth-token@3.3.2
  │ │ │ └─┬ rc@1.2.8
  │ │ │   ├── deep-extend@0.6.0
  │ │ │   ├── ini@1.3.5
  │ │ │   ├── minimist@1.2.0
  │ │ │   └── strip-json-comments@2.0.1
  │ │ └── registry-url@3.1.0
  │ ├── path-exists@3.0.0
  │ ├── read-cmd-shim@1.0.1
  │ ├─┬ read-pkg@3.0.0
  │ │ ├─┬ normalize-package-data@2.4.0
  │ │ │ ├─┬ is-builtin-module@1.0.0
  │ │ │ │ └── builtin-modules@1.1.1
  │ │ │ └─┬ validate-npm-package-license@3.0.3
  │ │ │   ├─┬ spdx-correct@3.0.0
  │ │ │   │ └── spdx-license-ids@3.0.0
  │ │ │   └─┬ spdx-expression-parse@3.0.0
  │ │ │     └── spdx-exceptions@2.1.0
  │ │ └── path-type@3.0.0
  │ ├── safe-buffer@5.1.2
  │ ├── semver@5.5.0
  │ ├── signal-exit@3.0.2
  │ ├── slash@1.0.0
  │ ├─┬ strong-log-transformer@1.0.6
  │ │ ├── byline@5.0.0
  │ │ ├── duplexer@0.1.1
  │ │ ├── minimist@0.1.0
  │ │ └── moment@2.22.2
  │ ├─┬ temp-write@3.4.0
  │ │ ├── make-dir@1.3.0
  │ │ ├── temp-dir@1.0.0
  │ │ └── uuid@3.2.1
  │ ├─┬ write-file-atomic@2.3.0
  │ │ └── imurmurhash@0.1.4
  │ ├─┬ write-json-file@2.3.0
  │ │ ├── detect-indent@5.0.0
  │ │ └─┬ sort-keys@2.0.0
  │ │   └── is-plain-obj@1.1.0
  │ ├── write-pkg@3.2.0
  │ └─┬ yargs@8.0.2
  │   ├── camelcase@4.1.0
  │   ├─┬ cliui@3.2.0
  │   │ ├─┬ string-width@1.0.2
  │   │ │ └── is-fullwidth-code-point@1.0.0
  │   │ └─┬ wrap-ansi@2.1.0
  │   │   └─┬ string-width@1.0.2
  │   │     └── is-fullwidth-code-point@1.0.0
  │   ├── decamelize@1.2.0
  │   ├── get-caller-file@1.0.2
  │   ├─┬ os-locale@2.1.0
  │   │ ├── execa@0.7.0
  │   │ ├─┬ lcid@1.0.0
  │   │ │ └── invert-kv@1.0.0
  │   │ └─┬ mem@1.1.0
  │   │   └── mimic-fn@1.2.0
  │   ├─┬ read-pkg-up@2.0.0
  │   │ └─┬ read-pkg@2.0.0
  │   │   ├─┬ load-json-file@2.0.0
  │   │   │ ├── parse-json@2.2.0
  │   │   │ └── pify@2.3.0
  │   │   └── path-type@2.0.0
  │   ├── require-directory@2.1.1
  │   ├── require-main-filename@1.0.1
  │   ├── which-module@2.0.0
  │   ├── y18n@3.2.1
  │   └─┬ yargs-parser@7.0.0
  │     └── camelcase@4.1.0
  └── rimraf@2.6.2
  
  > npm run build
  
  > @ build /tmp/pip-install-cg86t734/nbdime
  > lerna run build
  
  lerna info version 2.11.0
  lerna info versioning independent
  lerna ERR! build Errored while running script in 'nbdime-webapp'
  lerna ERR! execute callback with error
  lerna ERR! Error: Command failed: npm run build
  lerna ERR! sh: 1: webpack: not found
  lerna ERR!
  lerna ERR! npm ERR! Linux 4.4.0-128-generic
  lerna ERR! npm ERR! argv "/usr/bin/nodejs" "/usr/bin/npm" "run" "build"
  lerna ERR! npm ERR! node v4.2.6
  lerna ERR! npm ERR! npm  v3.5.2
  lerna ERR! npm ERR! file sh
  lerna ERR! npm ERR! code ELIFECYCLE
  lerna ERR! npm ERR! errno ENOENT
  lerna ERR! npm ERR! syscall spawn
  lerna ERR! npm ERR! nbdime-webapp@1.0.0 build: `webpack -p`
  lerna ERR! npm ERR! spawn ENOENT
  lerna ERR! npm ERR!
  lerna ERR! npm ERR! Failed at the nbdime-webapp@1.0.0 build script 'webpack -p'.
  lerna ERR! npm ERR! Make sure you have the latest version of node.js and npm installed.
  lerna ERR! npm ERR! If you do, this is most likely a problem with the nbdime-webapp package,
  lerna ERR! npm ERR! not with npm itself.
  lerna ERR! npm ERR! Tell the author that this fails on your system:
  lerna ERR! npm ERR!     webpack -p
  lerna ERR! npm ERR! You can get information on how to open an issue for this project with:
  lerna ERR! npm ERR!     npm bugs nbdime-webapp
  lerna ERR! npm ERR! Or if that isn't available, you can get their info via:
  lerna ERR! npm ERR!     npm owner ls nbdime-webapp
  lerna ERR! npm ERR! There is likely additional logging output above.
  lerna ERR!
  lerna ERR! npm ERR! Please include the following file with any support request:
  lerna ERR! npm ERR!     /tmp/pip-install-cg86t734/nbdime/packages/webapp/npm-debug.log
  lerna ERR!
  lerna ERR! > nbdime-webapp@1.0.0 build /tmp/pip-install-cg86t734/nbdime/packages/webapp
  lerna ERR! > webpack -p
  lerna ERR!
  lerna ERR!
  lerna ERR!     at /tmp/pip-install-cg86t734/nbdime/node_modules/execa/index.js:236:11
  lerna WARN complete Waiting for 1 child process to exit. CTRL-C to exit immediately.
  lerna ERR! build Errored while running script in 'nbdime'
  { [Error: Command failed: npm run build
  sh: 1: webpack: not found

```

**Comment by stas00** [comment]
`pip3 install -e git+https://github.com/jupyter/nbdime#egg=nbdime` failed right away not liking the default nodejs-4.2.6 coming with ubuntu 16.04. I followed https://www.digitalocean.com/community/tutorials/how-to-install-node-js-on-ubuntu-16-04 and installed nodejs-8.11.3.

Now, the above install command proceeds further and then fails in the exact same place as I reported above with identical errors.

I think it may have something to do with being run under root user (via sudo). Since I can successfully build/install from git via venv as a normal user.  Either there is some issue with that or the 2 environments differ and some dependencies are missing in the system-wide environment, but they aren't being picked by the build system. It does complain about webpack missing.

Nevertheless, now that I have a way of testing your master branch without it being released using venv, I'm fine with this issue not being resolved. So if you feel you'd like to resolve it, I'll follow your guidance and report back the advancement. If you'd rather do something different, I will stick with the build via venv which works.
